### ISSUE TITLE
feat: multi-turn OPD for agentic tool-use trajectories

### ISSUE THREAD

**Issue by Zhou-Weixin** [Algorineko/AgenticArXiv-RL#56]
## 背景

#45 与 #53 已加入基于 TRL GKDTrainer 的 OPD（On-Policy Distillation）训练路径。当前实现的边界在 README 和 `train_opd.py` 中已有说明：每条样本只蒸馏一个 ReAct 动作，不包含 `Action → Observation → Action` 的多轮环境交互。

对于本项目的 Agentic 工具调用场景，单轮 OPD 尚不能覆盖复合任务中的状态变化、工具反馈以及基于 Observation 的后续决策。因此希望在保留现有单轮实现的基础上，补充多轮 Agentic OPD。

## 目标

实现面向工具调用轨迹的多轮 OPD：

```text
学生生成 Action
→ MockArxivEnv 执行工具
→ Observation 写回上下文
→ 学生继续生成下一轮 Action
→ 教师对学生生成 token 提供蒸馏信号
→ 直到 FINISH、解析失败或达到 max_turns
```

训练仍采用 OPD 的教师逐 token 信号，不使用任务 reward 或 value model；可验证奖励仅用于 Canary、阶段验证和训练前后评测。

## 初步实现范围

- 学生模型进行多轮 on-policy rollout。
- 复用现有 `MockArxivEnv` / environment factory，每条 rollout 使用独立环境实例，避免会话状态串扰。
- 将工具 Observation 作为下一轮上下文，但通过 mask 排除在蒸馏 loss 之外。
- 仅对学生生成的 assistant token 计算 reverse-KL：
  `D_KL(π_student || π_teacher)`。
- 正确处理 `FINISH`、解析失败、工具执行失败和 `max_turns` 截断。
- 保留现有单轮 OPD 行为，避免破坏已有命令和实验可比性。
- 第一版可要求 teacher/student 使用兼容 tokenizer 与 vocabulary；不兼容时启动阶段明确报错，跨 tokenizer 蒸馏不在本 issue 范围内。
- 沿用现有 Canary、StageVerifier 和训练日志能力，并增加多轮轨迹健康度指标。
- 增加与 SFT、单轮 OPD、GRPO 的离线 benchmark 对比入口或文档说明。

## 建议测试

- Observation token 的 loss mask 恒为 0，assistant 生成 token 的 mask 正确。
- 教师只提供梯度目标且始终冻结，参数不会被更新。
- reverse-KL 方向保持为 `student || teacher`。
- 每条 rollout 的环境状态相互隔离。
- 多轮轨迹能根据真实 Observation 继续生成，而不是由模型自行编造 Observation。
- `FINISH`、parse error、tool error、force stop 的终止行为正确。
- 单轮默认路径保持向后兼容。
- tokenizer/vocabulary 不兼容时响亮失败，不静默产生错误 loss。

## 非目标

- 本 issue 暂不实现跨 tokenizer/vocabulary 的分布对齐。
- 暂不实现 API 教师；教师仍需提供本地逐 token logits/logprobs。
- 暂不引入 reward model、value model 或 PG-OPD。
- 暂不处理全异步 rollout/训练架构。

## 认领

这个方向由我认领实现。如果维护者认可上述范围，请将该 issue assign 给我（@Zhou-Weixin）。我会先提交包含整体接口、mask 语义和测试骨架的 Draft PR，再逐步补齐训练实现与文档。
